### PR TITLE
Implicit and explicit schemes implemented

### PR DESCRIPTION
**Pull request by apuigferrat**

### CODE REVIEW

**Comment by ipouplana** [comment]
Codacy suggested some changes... I think we should look into them before merging this PR.

**Review by ipouplana — DISMISSED** [review]
Thanks!

File: applications/FluidTransportApplication/python_scripts/apply_scalar_constraint_function_process.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import KratosMultiphysics.FluidTransportApplication as KratosFluidTransport
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return ApplyScalarConstraintFunctionProcess(Model, settings["Parameters"])
+
+## All the python processes should be derived from "python_process"
+
+class ApplyScalarConstraintFunctionProcess(KratosMultiphysics.Process):
```
Comment (philbucher) [inline applications/FluidTransportApplication/python_scripts/apply_scalar_constraint_function_process.py:11]: I am not familiar with this app but the functionality you implement here is already in the existing in the `AssignScalarVariableProcess` => you can specify a python function as input

Same applies for your other python-process

**Review by philbucher — COMMENTED** [review]
you added tetgen, travis won't let you merge before you remove it

File: applications/FluidTransportApplication/python_scripts/apply_scalar_constraint_function_process.py (hunk 1 above)
Comment (ipouplana) [reply applications/FluidTransportApplication/python_scripts/apply_scalar_constraint_function_process.py:11]: These two processes are ad-hoc implementations for 2 particular problems we had run...

**Review by ipouplana — COMMENTED** [review]
Thanks @philbucher , we didn't notice that tetgen was added... We will delete it tomorrow

**Review by ipouplana — APPROVED** [review]
Thanks!

**Pull request merged**